Zend/Optimizer/zend_func_info.c

```diff
@@ -96,18 +97,20 @@ ZEND_API int zend_func_info_rid = -1;
 
 uint32_t zend_get_internal_func_info(
 		const zend_function *callee_func, const zend_call_info *call_info, const zend_ssa *ssa) {
-	if (callee_func->common.scope) {
-		/* This is a method, not a function. */
+	if (callee_func->common.scope && !call_info->is_prototype) {
```

Comment:
```suggestion
	if (callee_func->common.scope && (!call_info->is_prototype || !callee_func->common.function_name)) {
```

With something like that, could this use get_function_or_method_name without the check for null/main?

Reply:
Good idea, but I think it would be more clear to introduce a new function which returns `null` instead of `main`, and `get_function_or_method_name()` could use this.